utils/service_discovery/sd_backend.py

```diff
@@ -0,0 +1,178 @@
+# std
```

Comment:
This whole module would benefit from better logging as it can be pretty tricky to debug otherwise